### DISCUSSION TITLE
Rework Organic Maps into a One Page Application

### DISCUSSION THREAD

**Discussion by Paspieds** [organicmaps/organicmaps#9317] [Ideas]
My proposition is to rework the application so at the end, one Main Activity remains hosting multiple fragments corresponding to the Search, Bookmarks, Setting, Download maps views (and others if I forgot one). At the end the User Flow will stay the same by navigating through the views thanks to the bottom navigation buttons and in the same order. This should make the navigation smoother and lighter avoiding heavy transitions between activities (also, fragments are a lot lighter than activities).

This is only a proposition. I don't have the necessary knowledge to claim that I know this is THE solution or even a priority.

**Comment by rtsisyk** [comment]
It is already almost like that. What problem are you trying to solve?

**Comment by vng** [comment]
Actually, we still have legacy Activity architecture, and if you switch to OpenGL renderer (type "?gl" in search and restart the app), you can see that the map scene is constantly invalidated when switching between Map and Search/Bookmarks/Settings.

@Paspieds I can't say for sure how difficult it is to migrate on Fragments and what issues may occur, especially with our hardware rendering (Vulkan or OpenGL) and current NDK interop (which is also hardcoded into Activities), but you definitely can make an experimental PR, at least for one simple Activity like Settings. And we will be glad to look at it.

**Reply by rtsisyk** [reply]
We have fragments already, but there are still some screens implemented in separate activities.

**Reply by Paspieds** [reply]
@vng very interesting, I'll definitely work on that matter and come back if somehow I found a satisfactory solution. I sure don't know (surely less than you) as well how much of a problem doing such a migration could be, but I think it is interesting to try it out since this could lead to improve the application.

**Comment by visacode** [comment]
where can I submit my design contribution? I have worked on a full revamp of the user experience and will love to have feedback and keep it up looking fresh.
